tests/integ_tests/worker_heartbeat_tests.rs

```diff
@@ -0,0 +1,322 @@
+    let mut worker = starter.worker().await;
+    let worker_instance_key = worker.worker_instance_key();
+
+    // Run a workflow
+    worker.register_wf(wf_name.to_string(), |ctx: WfContext| async move {
+        ctx.activity(ActivityOptions {
+            activity_type: "pass_fail_act".to_string(),
+            input: "pass".as_json_payload().expect("serializes fine"),
+            start_to_close_timeout: Some(Duration::from_secs(1)),
+            ..Default::default()
+        })
+        .await;
+        Ok(().into())
+    });
+    worker.register_activity("pass_fail_act", |_ctx: ActContext, i: String| async move {
+        Ok(i)
+    });
+
+    starter
+        .start_with_worker(wf_name.clone(), &mut worker)
+        .await;
+    worker.run_until_done().await.unwrap();
+
+    let client = starter.get_client().await;
+    let workers_list = client
+        .list_workers(100, Vec::new(), String::new())
+        .await
+        .unwrap();
+    // Since list_workers finds all workers in the namespace, must find specific worker used in this
+    // test
+    let worker_info = workers_list
+        .workers_info
+        .iter()
+        .find(|worker_info| {
+            if let Some(hb) = worker_info.worker_heartbeat.as_ref() {
+                hb.worker_instance_key == worker_instance_key.to_string()
+            } else {
+                false
+            }
+        })
+        .unwrap();
+    let heartbeat = worker_info.worker_heartbeat.as_ref().unwrap();
+    assert!(heartbeat.task_queue.starts_with(&wf_name));
+    assert_eq!(heartbeat.worker_identity, "integ_tester");
+    assert_eq!(heartbeat.sdk_name, "temporal-core");
+    assert_eq!(heartbeat.sdk_version, "0.1.0");
+    assert_eq!(heartbeat.status, WorkerStatus::Shutdown as i32);
+    assert!(within_two_minutes_ts(heartbeat.start_time.unwrap()));
+    assert!(within_two_minutes_ts(heartbeat.heartbeat_time.unwrap()));
+    assert!(within_duration(
+        heartbeat.elapsed_since_last_heartbeat.unwrap(),
+        Duration::from_secs(1)
+    ));
+
+    let workflow_poller_info = heartbeat.workflow_poller_info.unwrap();
+    assert!(!workflow_poller_info.is_autoscaling);
+    assert!(within_two_minutes_ts(
+        workflow_poller_info.last_successful_poll_time.unwrap()
+    ));
+    let sticky_poller_info = heartbeat.workflow_sticky_poller_info.unwrap();
+    assert!(!sticky_poller_info.is_autoscaling);
+    assert!(within_two_minutes_ts(
+        sticky_poller_info.last_successful_poll_time.unwrap()
+    ));
+    let nexus_poller_info = heartbeat.nexus_poller_info.unwrap();
+    assert!(!nexus_poller_info.is_autoscaling);
+    assert!(nexus_poller_info.last_successful_poll_time.is_none());
+    let activity_poller_info = heartbeat.activity_poller_info.unwrap();
+    assert!(!activity_poller_info.is_autoscaling);
+    assert!(within_two_minutes_ts(
+        activity_poller_info.last_successful_poll_time.unwrap()
+    ));
+
+    assert_eq!(heartbeat.total_sticky_cache_hit, 2);
+}
```

Comment:
We should check that all the various slot metrics are getting reported properly too